src/validation.cpp

```diff
@@ -1684,6 +1692,19 @@ bool GetReferral(const uint256 &hash, referral::ReferralRef &refOut, uint256 &ha
                     hash.GetHex().c_str(),
                     refOut->GetHash().GetHex().c_str());
 
+        //Find the height based on the block hash 
+        auto bi = mapBlockIndex.find(hashBlock);
+        if (bi != mapBlockIndex.end()) {
+            pindex = bi->second;
+        }
+
+        if(pindex && pindex->nHeight > 0) {
+            auto height = prefviewdb->GetReferralHeight(refOut->GetAddress());
+            if(height > 0) {
```

Comment:
it should be `height == 0`?

Reply:
no, in this case GetReferralHeight returns -1 if it isn't indexed in the DB. It's lazy indexed on demand. Which is reasonable since the height is immutable once in the chain.

Reply:
ok, then it should be `if (height == -1)`
why would we insert to cache if it's already there?